src/Data/Language.hs

```diff
@@ -137,3 +140,19 @@ textToLanguage = \case
   "TSX" -> TSX
   "PHP" -> PHP
   _ -> Unknown
+
+
+newtype PerLanguageModes = PerLanguageModes
+  { pythonMode :: LanguageMode
+  }
```

Comment:
Per @dcreager, representing this information per-language allows us to migrate individual languages one at a time.

I’m very content with that approach, but I’m not 100% happy with this implementation of it; nevertheless, it accomplishes it well enough for today, and I’m intending to make some simplifications and generalizations of it in a follow-up PR as we add support for more languages’ precise ASTs.